Cron/PendingOrdersUpdate.php

```diff
@@ -54,15 +54,24 @@ public function execute(): PendingOrdersUpdate
             $amwalOrderId = $order->getAmwalOrderId();
             $orderId = $order->getEntityId();
 
+            if (strpos($amwalOrderId, '-canceled') !== false) {
+                $this->logger->notice(
+                    sprintf('Skipping Order %s as it was canceled because the payment was retried.', $orderId)
+                );
+                continue;
+            }
+
```

Comment:
Looks good. I think we need similar logic in `Model\Data\AmwalOrderDetails.php` and `Model\Checkout\PayOrder.php`. I wonder if we need to refactor and include all of this in the `Model\Data\OrderUpdate.php` which is used by the three places.